### PR TITLE
README: Fix GHA badge and convert to Markdown

### PR DESCRIPTION
**Pull request by hugovk**
Fix the GitHub Actions badge URL, and add a Codecov coverage badge.

Convert the README to Markdown, we don't need the full complexity of reStructuredText here.

Also change "checkin" to "commit" to follow the familiar Git terminology.

### CODE REVIEW

**Comment by hugovk** [comment]
re: https://github.com/python/core-workflow/issues/329#issuecomment-2024019908

Before merge, let's make sure CLA bot is up and running, and that another core dev can merge.

**Comment by hugovk** [comment]
Close/re-open to check CLA bot ✅

File: README.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+## blurb subcommands
+
+Like many modern utilities, **blurb** has only one executable
+(called `blurb`), but provides a diverse set of functionality
+through subcommands.  The subcommand is the first argument specified
+on the command-line.
+
+If you're a CPython contributor, you probably don't need to use
+anything except `blurb add`--and you don't even need to specify
```
Comment (ezio-melotti) [inline README.md:59]: Markdown doesn't convert `--`/`---` to en/em dashes, and the resulting "`blurb add`--and" looks a bit off to me.  We could either replace it with an actual em dash, or at least add spaces (or both).  Pick one:
```suggestion
anything except `blurb add`—and you don't even need to specify
```
```suggestion
anything except `blurb add` — and you don't even need to specify
```
```suggestion
anything except `blurb add` -- and you don't even need to specify
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+is that?
+
+First, it makes naming the next version a late-binding decision.
+If we are currently working on 3.6.5rc1, but there's a zero-day
+exploit and we need to release an emergency 3.6.5 final, we don't
+have to fix up a bunch of metadata.
+
+Second, it means that if you cherry-pick a commit forward or
+backwards, you automatically pick up the `NEWS` entry too.  You
+don't need to touch anything up--the system will already do
```
Comment (ezio-melotti) [inline README.md:235]:
```suggestion
don't need to touch anything up—the system will already do
```
```suggestion
don't need to touch anything up — the system will already do
```
```suggestion
don't need to touch anything up -- the system will already do
```

**Pull request merged**